lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -137,9 +181,340 @@ public boolean contains(Candidate candidate) {
     return null;
   }
 
+  private void updatePartitionStorage(FactPartition part) throws LensException {
```

Comment:
Uh. this is just for backward compatibility as we are having multiple storages in a Partition.
Also, this is common operation so I have put this codeblock into a method.